### DISCUSSION TITLE
On using Asyncio

### DISCUSSION THREAD

**Discussion by BhuwanPandey** [fastapi/typer#864] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorials in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
Hey there,
I have file abcd.py where code looks like

from asyncio import run as aiorun
import typer

app = typer.Typer()

async def async_process():
    print("hello I am async")

@app.command()
def hello():
    print("I am Hello")


@app.command()
def async_handle():
    aiorun(async_process())

if __name__ == "__main__":
    app()

when I run 
python abcd.py async_handle it run successfully but

but when I changed code to

from asyncio import run as aiorun
import typer

app = typer.Typer()

async def async_process():
    print("hello I am async")


@app.command()
def async_handle():
    aiorun(async_process())

if __name__ == "__main__":
    app()

when i run abcd.py async_handle, it say
Got unexpected extra argument (async-handle)
```

### Description

..

### Operating System

Windows

### Typer Version

0.12.3

### Python Version

3.11

**Comment by YuriiMotov** [comment, marked as answer]
This is because if you have only one command, Typer makes some optimizations and you don't need to specify command to call it. So, it treats `async_handle` in `abcd.py async_handle` as an argument, thus it shows that error.

The solution is to add a callback as described here: https://typer.tiangolo.com/tutorial/commands/one-or-multiple/#one-command-and-one-callback

```py
from asyncio import run as aiorun
import typer

app = typer.Typer()

async def async_process():
    print("hello I am async")


@app.command()
def async_handle():
    aiorun(async_process())

@app.callback()
def callback():
    pass

if __name__ == "__main__":
    app()
```

And, your command name will actually be `async-handle` (with dash instead of underscore).

Try it:
```
$ python abcd.py async-handle
hello I am async
```
